cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2Cluster.java

```diff
@@ -0,0 +1,328 @@
+    protected static final String ENV_VAR_KAFKA_MIRRORMAKER_2_TLS_AUTH_CLUSTERS = "KAFKA_MIRRORMAKER_2_TLS_AUTH_CLUSTERS";
+    protected static final String ENV_VAR_KAFKA_MIRRORMAKER_2_TLS_AUTH_CERTS_CLUSTERS = "KAFKA_MIRRORMAKER_2_TLS_AUTH_CERTS_CLUSTERS";
+    protected static final String ENV_VAR_KAFKA_MIRRORMAKER_2_TLS_AUTH_KEYS_CLUSTERS = "KAFKA_MIRRORMAKER_2_TLS_AUTH_KEYS_CLUSTERS";
+    protected static final String ENV_VAR_KAFKA_MIRRORMAKER_2_SASL_PASSWORD_FILES_CLUSTERS = "KAFKA_MIRRORMAKER_2_SASL_PASSWORD_FILES_CLUSTERS";
+    protected static final String ENV_VAR_KAFKA_MIRRORMAKER_2_OAUTH_TRUSTED_CERTS = "KAFKA_MIRRORMAKER_2_OAUTH_TRUSTED_CERTS";
+    protected static final String ENV_VAR_KAFKA_MIRRORMAKER_2_OAUTH_CLIENT_SECRETS_CLUSTERS = "KAFKA_MIRRORMAKER_2_OAUTH_CLIENT_SECRETS_CLUSTERS";
+    protected static final String ENV_VAR_KAFKA_MIRRORMAKER_2_OAUTH_ACCESS_TOKENS_CLUSTERS = "KAFKA_MIRRORMAKER_2_OAUTH_OAUTH_ACCESS_TOKENS_CLUSTERS";
+    protected static final String ENV_VAR_KAFKA_MIRRORMAKER_2_OAUTH_REFRESH_TOKENS_CLUSTERS = "KAFKA_MIRRORMAKER_2_OAUTH_REFRESH_TOKENS_CLUSTERS";
+
+    protected static final String OAUTH_SECRETS_BASE_VOLUME_MOUNT = "/opt/kafka/oauth/";
+
+    private List<KafkaMirrorMaker2ClusterSpec> clusters;
+
+    /**
+     * Constructor
+     *
+     * @param namespace Kubernetes/OpenShift namespace where Kafka Connect cluster
+     *                  resources are going to be created
+     * @param cluster   overall cluster name
+     */
+    private KafkaMirrorMaker2Cluster(String namespace, String cluster, Labels labels) {
+        super(namespace, cluster, labels);
+        this.name = KafkaMirrorMaker2Resources.deploymentName(cluster);
+        this.serviceName = KafkaMirrorMaker2Resources.serviceName(cluster);
+        this.ancillaryConfigName = KafkaMirrorMaker2Resources.metricsAndLogConfigMapName(cluster);
+    }
+
+    /**
+     * Creates instance of KafkaMirrorMaker2Cluster from CRD definition.
+     *
+     * @param kafkaMirrorMaker2 The Custom Resource based on which the cluster model should be created.
+     * @param versions The image versions for MirrorMaker 2.0 clusters.
+     * @return The MirrorMaker 2.0 cluster model.
+     */
+    public static KafkaMirrorMaker2Cluster fromCrd(KafkaMirrorMaker2 kafkaMirrorMaker2, 
+                                                   KafkaVersion.Lookup versions) {
+        KafkaMirrorMaker2Cluster cluster = new KafkaMirrorMaker2Cluster(kafkaMirrorMaker2.getMetadata().getNamespace(),
+                kafkaMirrorMaker2.getMetadata().getName(),
+                Labels.fromResource(kafkaMirrorMaker2).withKind(kafkaMirrorMaker2.getKind()));
+        KafkaMirrorMaker2Spec spec = kafkaMirrorMaker2.getSpec();
+        cluster.setOwnerReference(kafkaMirrorMaker2);
+        cluster.setImage(versions.kafkaMirrorMaker2Version(spec.getImage(), spec.getVersion()));
```

Comment:
_Maybe_ you have this elsewhere and I've not spotted it anywhere, but do we need a check that the version of Kafka connect is one which supports MM2? Or does it just naturally fail with an understandable error message when used with a <2.4 version of Kafka?

Reply:
I have already updated the versions build to ensure that the cluster operator STRIMZI_KAFKA_MIRROR_MAKER_2_IMAGES env variable only has 2.4.0 in the list, but that doesn't stop someone editing the env variable in `050-Deployment-strimzi-cluster-operator.yaml`. Have added a check and a test.